contracts/mocks/ERC20CloneFactory.sol

```diff
@@ -19,15 +19,12 @@ contract ERC20CloneFactory is CloneFactory {
 
   address public masterAddress;
 
-  event CloneCreated(address cloneAddress);
-
-  constructor(address _masterAddress) public {
-    masterAddress = _masterAddress;
+  constructor() public {
+    masterAddress = new ERC20Mock(msg.sender, 0);
   }
 
   function createNew(uint256 amount) public returns (address) {
     address cloneAddress = createClone(masterAddress);
-    emit CloneCreated(cloneAddress);
     ERC20Mock(cloneAddress).mint(msg.sender, amount);
```

Comment:
It's interesting that we're able to replicate the contents of the constructor with this call, but it's only available because this is a mock, were we need to expose that internal method: I wouldn't want someone to look at this and think this is good code they should copy.

I'm actually not sure how we should handle this, since we can't even use `ERC20Mintable` because there will be no minters. @frangio WDYT? Maybe test this with something that is constructor-less?

Reply:
Hey @nventuro I'm not sure if I correctly understood your thoughts about replication the contents of constructor but I believe you meant that it's wrong example for usage. What you would say to create a simple contract only for purpose of test like Peter's did [here](https://github.com/optionality/clone-factory/blob/master/test/Thing.sol) ?

Reply:
I've just hit on <SimpleToken> - how do you think @nventuro @frangio ? Could it be utilized as usage example?